### PR TITLE
ENH: update pivot op

### PR DESCRIPTION
**Pull request by Hank0626**
## What do these changes do?
- support one chunk simplification
- add shape infer and combine op

- [x] tests added / passed (if needed)
- [x] Ensure all linting tests pass

### CODE REVIEW

**Comment by UranusSeven** [comment]
Hi, could you provide the reasons behind these changes? Like what is the problem you are trying to solve?

**Comment by Hank0626** [comment]
> Hi, could you provide the reasons behind these changes? Like what is the problem you are trying to solve?

1. Like pivot_table op, pivot will have map, reduce and shuffle operations during computation. After the shuffle operation, it is possible that the first chunk does not contain any data (thus has the wrong shape), but during the test, when judging the shape, we use the shape of the first chunk to infer the shape of the result. So I added yield for inferring the shape and combine to fuse the final result with the ending.
2. I added the function `tile_one_chunk` to deal with the case of only one chunk, so that it can be more efficient to deal with the case of no need to split the chunk

**Comment by aresnow1** [comment]
Optimization is possible here. After yield, you can check if the chunk's columns are the same as the `output_columns`. If so, you can skip the combine phase, which can reduce some calculations.

**Comment by Hank0626** [comment]
> Optimization is possible here. After yield, you can check if the chunk's columns are the same as the `output_columns`. If so, you can skip the combine phase, which can reduce some calculations.

done, also done for pivot_table https://github.com/xprobe-inc/xorbits/pull/485

File: python/xorbits/_mars/dataframe/base/tests/test_base_execution.py [hunk 1]
```diff
@@ -698,40 +698,48 @@ def sorted_df(df):
             "qux": [10, 20, 30, 40, 50, 60],
         }
     )
-    df = from_pandas_df(df_raw, chunk_size=2)
+    df = from_pandas_df(df_raw, chunk_size=1)
 
     # test basic pivot
     r = df.pivot(index="foo", columns="bar", values="baz")
-    result = r.execute().fetch()
+    result = r.execute(extra_config={"check_dtypes": False}).fetch()
```
Comment (aresnow1) [inline python/xorbits/_mars/dataframe/base/tests/test_base_execution.py:705]: Why skip checking dtypes?

Comment (Hank0626) [reply python/xorbits/_mars/dataframe/base/tests/test_base_execution.py:705]: deal with the order of dtypes

**Review by aresnow1 — APPROVED** [review]
LGTM

**Review by UranusSeven — APPROVED** [review]
LGTM. Good job :)

**Pull request merged**